cmsimple/classes/LinkChecker.php

```diff
@@ -123,14 +124,30 @@ private function countLinks(array $links)
      */
     public function determineLinkStatus(Link $link)
     {
+        global $cf;
+
         $parts = parse_url($link->getURL());
         if (isset($parts['scheme'])) {
             switch ($parts['scheme']) {
                 case 'http':
                     $status = $this->checkExternalLink($parts);
                     break;
+                case 'https':
+                    $status = $this->checkExternalLink($parts);
+                    break;
                 case 'mailto':
-                    $status = Link::STATUS_MAILTO;
+                    if (!empty($cf['link']['mailto'])) {
```

Comment:
Vielleicht besser auf `empty()` verzichten, siehe https://beberlei.de/2021/02/19/when_to_use_empty_in_php_i_say_never.html.

Reply:
> Vielleicht besser auf `empty()` verzichten, siehe

Na, ganz so eng muss man das sicher nicht sehen.
Aber ok, schauen wir erst mal, ob tatsächlich alles so läuft wie es soll, dann kann ich empty immer noch ersetzen.
Bei Zeile 139 oder 146 kann man sich auch direkt abfragen:
`if ($cf['link']['mailto'] == 'true'`
Die Variablen kommen aus config.php und sind von daher auf jeden Fall deklariert.

Reply:
Oh, ehrlich gesagt ist `empty()` beim Abfragen von Konfigurationsvariablen vermutlich genau das richtige (wir akzeptieren i.d.R. wohl noch immer `"0"` als `false`). Aber ich frage mich, warum du überhaupt auf die Konfigurationsvariable prüfst; sollten wir nicht beliebige mailto Links unterstützen, selbst wenn das Kontaktformular gar nicht aktiviert ist – oder gerade dann?

Reply:
Aber mehr als ein Hinweis kommt bei mailto oder tel nicht heraus. Prüfen lässt sich da nichts, max. könnte man prüfen ob eine korrekte E-Mail-Adresse oder Nummer angegeben ist.
Momentan ist der einzige Vorteil, man findet die Links, ohne jede Seite einzeln ansehen zu müssen.
Durch Feedback weiss ich, nicht jeder will diese Hinweise haben, deshalb konfigurierbar.

Mit 301 / 302 bei den Links ist es ebenso.
Mancher will nur wissen, wenn es am Ende wirklich 4xx oder 5xx ist.
Ob dazwischen eine Weiterleitung ist eben nicht.
Deswegen auch hier konfigurierbar.

Herauskommen soll eine reine (möglichst kurze) Fehlerliste.
Wenn ich jetzt mal den Standardcontent prüfen lasse bekomme ich eine ellenlange Liste, die nur schwer zu überblicken ist. Da gehen Fehler unter.

Im Moment gibt es bei alten Wiki-Links, durch einen Umleitungsfehler, „glücklicherweise“ zwei aufeinanderfolgende 301, so kann man das ganz gut testen.

Beides sollte aber korrigiert werden, sowohl die veralteten Links, als natürlich auch der Fehler in der Umleitung.

Reply:
Ah, ich dachte es würde `$cf['mailform']['email']` geprüft; eigene Konfigoptionen sind natürlich nicht verkehrt.